client/my-sites/plans/jetpack-plans/plans-filter-bar/get-highest-annual-discount.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { createSelector } from '@automattic/state-utils';
+
+/**
+ * Internal dependencies
+ */
+import { AppState } from 'calypso/types';
+import { getProductCost } from 'calypso/state/products-list/selectors/get-product-cost';
+import { getCurrentUserCurrencyCode } from 'calypso/state/current-user/selectors';
+import { getMonthlySlugFromYearly } from '../convert-slug-terms';
+
+const getAnnualDiscount = ( state: AppState, annualProductSlug: string ) => {
+	const yearly = getProductCost( state, annualProductSlug );
+
+	const monthlyProductSlug = getMonthlySlugFromYearly( annualProductSlug );
+	const monthly = monthlyProductSlug && getProductCost( state, monthlyProductSlug );
+
+	// Protect against null values and division by zero
+	if ( ! yearly || ! monthly ) {
+		return null;
+	}
+
+	const monthlyCostPerYear = monthly * 12;
+	const yearlySavings = monthlyCostPerYear - yearly;
+
+	return yearlySavings / monthlyCostPerYear;
+};
```

Comment:
This function is the only change in logic inside this refactor PR -- I moved the logic to calculate the discount for an individual slug into its own function, for readability sake.